libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java

```diff
@@ -150,6 +157,22 @@ public UserIdentity generateUserIdentity(String userId, String sub, String vot)
             Optional<JsonNode> drivingPermitClaim =
                     generateDrivingPermitClaim(successfulVCStoreItems);
             drivingPermitClaim.ifPresent(userIdentityBuilder::drivingPermitClaim);
+            if (configService.enabled(EXIT_CODES)) {
+                userIdentityBuilder.exitCode(
+                        contraIndicators.ciCodesToAlwaysRequiredExitCodes(
+                                configService.getContraIndicatorConfigMap(),
+                                configService.getSsmParameter(ALWAYS_REQUIRED_EXIT_CODES)));
```

Comment:
I can't quite work out what the right answer is, but there seems to be something fishy about our abstractions - passing in the config to a method like this 🤔 

Should we actually have the `toExitCodes` methods in the `UserIdentityService` or `CimitService`, and just use `contraIndicators.getCiCodes()` here? Similarly with the calls below.

Maybe I'm used to a much more functional view of the world, and not as used to having model classes with business logic!

Reply:
Yeah. I went back and forth on this a few times. Originally I was going to just pass in the config service, but that felt too tightly coupled. But I also don't love passing this stuff in.

The `ContraIndicators` class is more of a wrapper around a map, that extends its capabilities. The actual model is the `ContraIndicator` class that the map contains. I agree it's a bit weird.

Not sure I'd want to use the CimitService to do this, it's more about interfacing with the Cimit component.

I'll move the conversion to exit codes into the user identity service, see how it looks.

Reply:
Have pulled the `toExitCode` methods in now if you want to take another look.